flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskStopTest.java

```diff
@@ -1,157 +0,0 @@
-import org.apache.flink.runtime.jobgraph.JobVertexID;
-import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
-import org.apache.flink.runtime.jobgraph.tasks.InputSplitProvider;
-import org.apache.flink.runtime.jobgraph.tasks.StoppableTask;
-import org.apache.flink.runtime.memory.MemoryManager;
-import org.apache.flink.runtime.metrics.groups.TaskIOMetricGroup;
-import org.apache.flink.runtime.metrics.groups.TaskMetricGroup;
-
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
-
-import java.lang.reflect.Field;
-import java.util.Collections;
-import java.util.concurrent.Executor;
-
-import scala.concurrent.duration.FiniteDuration;
-
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
-@RunWith(PowerMockRunner.class)
-@PrepareForTest({ TaskDeploymentDescriptor.class, JobID.class, FiniteDuration.class })
-public class TaskStopTest {
-	private Task task;
-
-	public void doMocking(AbstractInvokable taskMock) throws Exception {
-
-		TaskInfo taskInfoMock = mock(TaskInfo.class);
-		when(taskInfoMock.getTaskNameWithSubtasks()).thenReturn("dummyName");
-
-		TaskManagerRuntimeInfo tmRuntimeInfo = mock(TaskManagerRuntimeInfo.class);
-		when(tmRuntimeInfo.getConfiguration()).thenReturn(new Configuration());
-
-		TaskMetricGroup taskMetricGroup = mock(TaskMetricGroup.class);
-		when(taskMetricGroup.getIOMetricGroup()).thenReturn(mock(TaskIOMetricGroup.class));
-
-		BlobCacheService blobService =
-			new BlobCacheService(mock(PermanentBlobCache.class), mock(TransientBlobCache.class));
-
-		task = new Task(
-			mock(JobInformation.class),
-			new TaskInformation(
-				new JobVertexID(),
-				"test task name",
-				1,
-				1,
-				"foobar",
-				new Configuration()),
-			mock(ExecutionAttemptID.class),
-			mock(AllocationID.class),
-			0,
-			0,
-			Collections.<ResultPartitionDeploymentDescriptor>emptyList(),
-			Collections.<InputGateDeploymentDescriptor>emptyList(),
-			0,
-			mock(TaskStateSnapshot.class),
-			mock(MemoryManager.class),
-			mock(IOManager.class),
-			mock(NetworkEnvironment.class),
-			mock(BroadcastVariableManager.class),
-			mock(TaskManagerActions.class),
-			mock(InputSplitProvider.class),
-			mock(CheckpointResponder.class),
-			blobService,
-			mock(LibraryCacheManager.class),
-			mock(FileCache.class),
-			tmRuntimeInfo,
-			taskMetricGroup,
-			mock(ResultPartitionConsumableNotifier.class),
-			mock(PartitionProducerStateChecker.class),
-			mock(Executor.class));
-		Field f = task.getClass().getDeclaredField("invokable");
-		f.setAccessible(true);
-		f.set(task, taskMock);
-
-		Field f2 = task.getClass().getDeclaredField("executionState");
-		f2.setAccessible(true);
-		f2.set(task, ExecutionState.RUNNING);
-	}
-
-	@Test(timeout = 20000)
-	public void testStopExecution() throws Exception {
```

Comment:
This only tested that stop is invoked. Should be covered by `testSetsUserCodeClassLoader`.